docs/markdown/Rewriter.md

```diff
@@ -0,0 +1,183 @@
+---
+short-description: Automatic modification of the build system files
+...
+
+# Meson file rewriter
+
+Since version 0.50.0, meson has the functionality to perform some basic
+modification on the `meson.build` files from the command line. The currently
+supported operations are:
+
+- For build targets:
+  - Add/Remove source files
+  - Add/Remove targets
+  - Modify a select set of kwargs
+  - Print some JSON information
+- For dependencies:
+  - Modify a select set of kwargs
+- For the project function:
+  - Modify a select set of kwargs
+  - Modify the default options list
+
+The rewriter has both, a normal command line interface and a "script mode". The
+normal CLI is mostly designed for everyday use. The "script mode", on the
+other hand, is meant to be used by external programs (IDEs, graphical
+frontends, etc.)
+
+## Using the rewriter
+
+All rewriter functions are accessed via `meson rewrite` (or the rewriter alias
+`meson rw`). The meson rewriter assumes that it is run inside the project root
```

Comment:
Can we remove this alias? It seems rewriting is really not something that is done a lot from the command line that needs very short commands.

Reply:
I would argue that a `meson rw tgt <target name> add src/*.cpp` would be called fairly often.

Reply:
That's a fun command line for sure. But to be useful it would need to detect sources already in the list and not duplicate those.

i'm still not sure adding this should alias is a good thing. Because it's quite cryptic. I'd try to see if tab completion would not be enough for interactive usage.

Reply:
Ah, I forgot to handle duplicates... Making a quick PR for this...

Also, yes it is cryptic, but you don't have to use this since these are only aliases. This is meant for users that already know the syntax of the rewriter. I also don't know about tab completion. Have never written bash / zsh completion scripts, so it could be a while before they are in meson.

Reply:
i'd prefer to keep the long version only for now because meson command line semantics are already very complicated and we really should try to make our lifes as easy as possible when we will need to clean the whole command line story up in the future.

Reply:
Seeing the problems with the introspection API, I guess I can't really argue with that then...

Reply:
Done